source/fundamentals/typescript.txt

```diff
@@ -79,81 +84,119 @@ You can find a code snippet that shows how to specify a type for the ``FindCurso
+preceding classes, see the :ref:`<node-driver-limitations-mutual-recursion>` section.
 
-.. _node-driver-typescript-limitations-dot-notation:
+.. _node-driver-limitations:
 
-The driver cannot infer the type of values with keys containing **dot
-notation**. Dot notation is a property access syntax for navigating BSON objects.
-Click on the tabs to see code snippets that highlight this behavior:
+Limitations For Driver Version {+version+}
+----------------------------------
 
-.. tabs::
+The following subsections describe the TypeScript specific limitations of
+version {+version+} of the {+driver-long+}. 
 
-  .. tab:: Dot Notation
-     :tabid: dot-notation
+Many limitations of the {+driver-short+} relate to **recursive types**.
+A recursive type is a type that references itself. You can update
+the :ref:`Pet <mongodb-node-typescript-pet-interface>` interface
+to be recursive by allowing a pet to have its own pet. The following is the
+recursive ``Pet`` interface:
 
-     The following code snippet does not raise a type error:
+.. _node-driver-limitations-recursive-pet:
 
-     .. literalinclude:: /code-snippets/typescript/dot-notation.ts
-        :language: typescript
-        :linenos:
-        :start-after: start-no-error
-        :end-before: end-no-error
+.. code-block:: typescript
+   :emphasize-lines: 2
 
-  .. tab:: Nested Objects
-     :tabid: nested-objects
+   interface RecursivePet {
+      pet?: RecursivePet;
+      name: string;
+      age: number;
+   }
 
-     The following code snippet raises a type error:
+Recursive Types and Dot Notation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-     .. literalinclude:: /code-snippets/typescript/dot-notation.ts
-        :language: typescript
-        :linenos:
-        :start-after: start-error
-        :end-before: end-error
+The {+driver-short+} cannot provide type safety within nested instances of
+recursive types referenced through **dot notation**. Dot notation is a property
+access syntax for navigating BSON objects.
 
-     This is the error:
+For example, the following code snippet references a nested instance of the
+:ref:`RecursivePet <node-driver-limitations-recursive-pet>` interface
+with an incorrect type using dot notation, but the TypeScript compiler
+does not raise an error:
 
-     .. code-block:: text
+.. code-block:: typescript
+   :emphasize-lines: 3
 
-        Type 'string' is not assignable to type 'number'.
+   database
+      .collection<RecursivePet>("your-collection")
+      .findOne({ "pet.age": "Spot" });
 
-Despite the lack of type safety, we still recommend that you use dot notation to
-access nested fields in query and update documents when you use TypeScript. You
-must manually check that your nested field values have your intended type.
+The following code snippet references a top-level instance of the
+``RecursivePet`` interface with an incorrect type and raises a type error:
 
-.. note:: Reason To Use Dot Notation
+.. tabs::
```

Comment:
Wouldn't it be better to be able to show both code snippet and error without having to select the tab?

Reply:
Apologies, still trying to break my tabs habit.